components/x-article-save-button/src/ArticleSaveButton.jsx

```diff
@@ -0,0 +1,52 @@
+import { h } from '@financial-times/x-engine';
+import articleSaveStyles from './ArticleSaveButton.scss';
+import classNames from 'classnames';
+
+export const ArticleSaveButton = props => {
+	const getLabel = props => {
+		if (props.saved) {
+			return 'Saved to myFT';
+		}
+
+		return props.contentTitle ? `Save ${props.contentTitle} to myFT for later` : 'Save this article to myFT for later';
+	};
+
+	return (
+		<form
+			className={classNames(articleSaveStyles.root)}
+			action={`/myft/save/${props.contentId}`}
+			method="GET"
+			data-content-id={props.contentId}
+			onSubmit={event => {
+				event.preventDefault();
+				const detail = {
+					action: props.saved ? 'remove' : 'add',
+					actorType: 'user',
+					relationshipName: 'saved',
+					subjectType: 'content',
+					subjectId: props.contentId,
+					token: props.csrfToken
+				};
+
+				event.target.dispatchEvent(new CustomEvent('x-article-save-button', { bubbles: true, detail }));
```

Comment:
It would be simpler for apps consuming this component within React if, instead of dispatching an event, this called a prop on `ArticleSaveButton` (e.g. `onSaveClick` or similar). If the prop isn't there, it could fall back to dispatching the event.

```suggestion
				if(typeof props.onSaveClick === 'function') {
					props.onSaveClick(detail);
				} else {
					event.target.dispatchEvent(new CustomEvent('x-article-save-button', { bubbles: true, detail }));
				}
```

This way, React consumers won't have to worry about lifecycles and attaching event handlers.